sdk/core/System.ClientModel/src/Convenience/OptionalClientResultOfT.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+
+namespace System.ClientModel;
+
+public class OptionalClientResult<T> : ClientResult
+{
+    private readonly T? _value;
+
+    protected internal OptionalClientResult(T? value, PipelineResponse response) : base(response)
+        => _value = value;
+
+    /// <summary>
+    /// Gets the value returned by the service. Accessing this property will throw if <see cref="HasValue"/> is false.
+    /// </summary>
+    public virtual T? Value => _value;
+
+    /// <summary>
+    /// Gets a value indicating whether the current instance has a valid value of its underlying type.
```

Comment:
I'm not sure what "valid" means in this context, so perhaps we should reflect the specific logic here?
```suggestion
    /// Gets a value indicating whether the current instance has a non-null value.
```

Reply:
I think I had copied what we have in Azure.Core today, but I will double check!  Happy to change it to something preferred either way.

Reply:
Updated!

Reply:
Ah, I see your point too now that we have taken "Nullable" out of the type name!